src/components/ui/members/add-member-dialog.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import type React from "react";
+
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { useUserMutation } from "@/lib/api/organizations/users";
+import { User, NewUser } from "@/types/user";
+import { useOrganizationStateStore } from "@/lib/providers/organization-state-store-provider";
+
+interface AddMemberDialogProps {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+}
+
+export function AddMemberDialog({ open, onOpenChange }: AddMemberDialogProps) {
+  const currentOrganizationId = useOrganizationStateStore(
+    (state) => state.currentOrganizationId
+  );
+
+  const { createNested: createUserNested } = useUserMutation(
+    currentOrganizationId
+  );
+  const [formData, setFormData] = useState({
+    firstName: "",
+    lastName: "",
+    displayName: "",
+    email: "",
+  });
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = e.target;
+    setFormData((prev) => ({
+      ...prev,
+      [name]: value,
+    }));
+  };
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+
+    const newUser: NewUser = {
+      first_name: formData.firstName,
+      last_name: formData.lastName,
+      display_name: formData.displayName,
+      email: formData.email,
+    };
+
+    await createUserNested(currentOrganizationId, newUser);
```

Comment:
You may want to import and consider calling `refresh()` after adding this, you'll get any other currently rendered list of Users automatically re-reading from the cache then.

Take a look at `addAction()` in actions-list.tsx for an example.